Source/JavaScriptCore/b3/air/AirAllocateRegistersByGraphColoring.cpp

```diff
@@ -1734,7 +1734,7 @@ class ColoringAllocator : public AllocatorType<IndexType, InterferenceSet, bank>
         // Note that the input property requires an analysis over ZDef's, so it's only valid so long
         // as the input gets a register. We don't know if the input gets a register, but we do know
         // that if it doesn't get a register then we will still emit this Move32.
-        if (inst.kind.opcode == Move32) {
+        if (inst.kind.opcode == Move32 && !is32Bit()) {
```

Comment:
ditto

Reply:
Same logic as in the other allocator--we don't need Move32 for its ZD-ness on 32-bit platforms, we can treat it as an ordinary move and we don't need this pessimization